src/commitments/kzg/srs_reader.rs

```diff
@@ -0,0 +1,170 @@
+use std::{
+    fs::File,
+    io::{BufRead, BufReader, Lines},
+};
+
+use crate::math::{
+    cyclic_group::IsGroup,
+    elliptic_curve::{
+        short_weierstrass::{
+            curves::bls12_381::{
+                curve::BLS12381Curve,
+                field_extension::{BLS12381PrimeField, Degree2ExtensionField},
+                twist::BLS12381TwistCurve,
+            },
+            point::ShortWeierstrassProjectivePoint,
+        },
+        traits::IsEllipticCurve,
+    },
+    field::element::FieldElement,
+};
+
+use super::{error::KzgError, G1, G2, SRS};
+
+macro_rules! check {
+    ($cond:expr) => {
+        ($cond).then_some(()).ok_or_else(|| {
+            dbg!($cond);
+            KzgError::BadArgs
+        })
+    };
+    ($one:expr, $other:expr) => {
+        ($one == $other).then_some(()).ok_or_else(|| {
+            dbg!($one);
+            dbg!($other);
+            KzgError::BadArgs
+        })
+    };
+}
+
+// TODO: This should be an env var
+const FIELD_ELEMENTS_PER_BLOB: usize = 4;
+const TRUSTED_SETUP_NUM_G1_POINTS: usize = FIELD_ELEMENTS_PER_BLOB;
+// TODO: change this, this only applies to Plonk
+const TRUSTED_SETUP_NUM_G2_POINTS: usize = 2;
+const BYTES_PER_G1: usize = 48;
+const BYTES_PER_G2: usize = 96;
+
+/// Load trusted setup from a file.
+///
+/// @remark The file format is `n1 n2 g1_1 g1_2 ... g1_n1 g2_1 ... g2_n2` where
+///     the first two numbers are in decimal and the remainder are hexstrings
+///     and any whitespace can be used as separators.
+///
+/// @remark See also load_trusted_setup().
+/// @remark The input file will not be closed.
+///
+/// @param[out] out Pointer to the loaded trusted setup data
+/// @param[in]  in  File handle for input
+pub fn load_trusted_setup_file(input: &mut File) -> Result<SRS, KzgError> {
```

Comment:
We need this to follow the C signature for ABI compatibility. Check how it's done [here](https://github.com/lambdaclass/lambdaworks_kzg/blob/main/src/lib.rs#L177-L191).